napari/components/dims.py

```diff
@@ -269,7 +269,7 @@ def indices(self):
                     p = np.clip(
                         self.point[axis],
                         np.round(self.range[axis][0]),
-                        np.round(self.range[axis][1]) - 1,
+                        np.round(self.range[axis][1]) - self.range[axis][2],
```

Comment:
```suggestion
                        np.round(self.range[axis][1] - self.range[axis][2]),
```
I think the rounding should happen after the step size is subtracted no? As @jni mentions this will be improved, but happy to fix a bug with what looks like backwards compatible behaviour for the majority of cases (where the range value is 1)

Reply:
hm. When I rethink this I do not understand why there is round here. I remove it and add tests which point what I expect. 

It is really important in context of reading metadata form files and set scaling in physical units like in #1384  

I finally understand why I need to multiply image voxels size by 10**9 To reduce IndexErrors. in napari. 

I vote to remove round. There is conversion to int bellow.